python/cudf/cudf/io/json.py

```diff
@@ -1,35 +1,101 @@
 # Copyright (c) 2019-2024, NVIDIA CORPORATION.
+from __future__ import annotations
 
+import os
 import warnings
 from collections import abc
 from io import BytesIO, StringIO
+from typing import TYPE_CHECKING, Any, Literal
 
 import numpy as np
 import pandas as pd
 
+import pylibcudf as plc
+
 import cudf
-from cudf._lib import json as libjson
+from cudf._lib.column import Column
+from cudf._lib.types import dtype_to_pylibcudf_type
+from cudf._lib.utils import _data_from_columns, data_from_pylibcudf_io
+from cudf.core.buffer import acquire_spill_lock
 from cudf.utils import ioutils
 from cudf.utils.dtypes import _maybe_convert_to_default_type
 
+if TYPE_CHECKING:
+    from cudf.core.column import ColumnBase
+
+
+def _add_df_col_struct_names(
```

Comment:
I think this will also used in `parquet.py`, should we put it in a util module?

Reply:
Ah good point. Sure I'll move this in a util module